requirements.txt

```diff
@@ -1,16 +1,18 @@
+async-lru
 backoff
 datasets
+diskcache
+httpx
 joblib~=1.3
 litellm<=1.49.1
+magicattr~=0.1.6
 openai
 optuna
 pandas
 pydantic~=2.0
 regex
 requests
 structlog
+tenacity
```

Comment:
why  is this here?

Reply:
Because when trying to run that test file you'd added it said tenacity wasn't installed

Reply:
?? Wild. Not sure what happened there

Reply:
Yeah lmao